### ISSUE TITLE
`??=` fails to promote variable whose type is a nullable type parameter

### ISSUE THREAD

**Issue by stereotype441** [dart-lang/language#3057]

**Comment by lrhn** [comment]
Indeed. We already say that if `T` is a type-of-interest, so is, I guess, `NonNull(T)`.

Which takes us from `T?`, which is definitely nullable, directly to `T&Object`, which is definitely non-nullable

We don't cover _potentially_ nullable, which is what `T` itself is. And which only type variables with nullable bounds can be.

Or will probably just be a slight tweak to finding related types of interest when the type is a type variable, or intersection type of one. Seems doable, and not too dangerous.
